src/test/java/net/snowflake/client/jdbc/FileUploaderSessionlessTest.java

```diff
@@ -279,12 +285,42 @@ public void testGetFileTransferMetadatasGCS() throws Exception {
     Assert.assertEquals("orders_100.csv", metadata.getPresignedUrlFileName());
   }
 
+  @Test
+  public void testGetFileTransferMetadataGCSWithUseRegionalUrl() throws Exception {
+    List<SnowflakeFileTransferMetadata> metadataList =
+        SnowflakeFileTransferAgent.getFileTransferMetadatas(exampleGCSJsonNodeWithUseRegionalUrl);
+    Assert.assertEquals(1, metadataList.size());
+
+    SnowflakeFileTransferMetadataV1 metadata =
+        (SnowflakeFileTransferMetadataV1) metadataList.get(0);
+
+    StageInfo stageInfo = metadata.getStageInfo();
+
+    assertTrue(stageInfo.getUseRegionalUrl());
+    assertEquals(Optional.of("storage.us-west1.rep.googleapis.com"), stageInfo.gcsCustomEndpoint());
+  }
+
+  @Test
+  public void testGetFileTransferMetadataGCSWithEndPoint() throws Exception {
+    List<SnowflakeFileTransferMetadata> metadataList =
+        SnowflakeFileTransferAgent.getFileTransferMetadatas(exampleGCSJsonNodeWithEndPoint);
+    Assert.assertEquals(1, metadataList.size());
+
+    SnowflakeFileTransferMetadataV1 metadata =
+        (SnowflakeFileTransferMetadataV1) metadataList.get(0);
+
+    StageInfo stageInfo = metadata.getStageInfo();
+
+    assertFalse(stageInfo.getUseRegionalUrl());
+    assertEquals(Optional.of("example.com"), stageInfo.gcsCustomEndpoint());
+  }
+
   @Test
   public void testGetFileTransferMetadatasUploadError() throws Exception {
     JsonNode downloadNode = mapper.readTree("{\"data\": {\"command\": \"DOWNLOAD\"}}");
     try {
       SnowflakeFileTransferAgent.getFileTransferMetadatas(downloadNode);
-      Assert.assertTrue(false);
+      assertTrue(false);
```

Comment:
Or maybe `fail()`? We don't have `assertThrows()`, right?

Reply:
we have `assertThrows` but I only changed the import to static and I haven't plan to rewrite all the tests
with such rewrite I would wait for junit5 to be ready
for now only changin `assertTrue(false)` to `fail()`